File: 2018/1/main.nim
import strutils, sequtils, math, sets

let nums = readFile("input").splitWhitespace.map(parseInt)
echo "A: ", nums.sum

proc b(nums: seq[int]): int =
  var f = 0
  var seen: HashSet[int]
  seen.init
  while true:
    for d in nums:
      seen.incl(f)
      f += d
      if seen.contains(f):
        return f
echo "B: ", b(nums)

File: 2018/1/main.py
#!/usr/bin/env python3

lines = [l.strip() for l in open('input') if l.strip()]
nums = [int(x[1:]) * (-1 if x[0] == '-' else 1) for x in lines]

def solveA(nums):
    return sum(nums)

def solveB(nums):
    seen = set([0])
    f = 0
    while True:
        for d in nums:
            f += d
            if f in seen:
                return f
            seen.add(f)
            
print('A', solveA(nums))
print('B', solveB(nums))
